src/components/multiSelect/MultiSelectBottomSheet.tsx

```diff
@@ -0,0 +1,232 @@
+import Touchable from 'src/components/Touchable'
+import Checkmark from 'src/icons/Checkmark'
+import Colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+const OPTION_HEIGHT = 60
+const MAX_OPTIONS_IN_VIEW = 5
+
+interface MultiSelectBottomSheetProps<T> {
+  forwardedRef: React.RefObject<GorhomBottomSheet>
+  onChange?: BottomSheetProps['onChange']
+  onClose?: () => void
+  onOpen?: () => void
+  handleComponent?: BottomSheetProps['handleComponent']
+  options: Option<T>[]
+  selectedOptions: T[]
+  setSelectedOptions: (selectedOptions: T[]) => void
+  selectAllText: string
+  title: string
+}
+
+export interface Option<T> {
+  id: T
+  text: string
+  iconUrl?: string
+}
+
+function MultiSelectBottomSheet<T extends string>({
+  forwardedRef,
+  onClose,
+  onOpen,
+  options,
+  setSelectedOptions,
+  selectedOptions,
+  selectAllText,
+  title,
+}: MultiSelectBottomSheetProps<T>) {
+  const { t } = useTranslation()
+  const scrollViewRef = useRef<ScrollView>(null)
+
+  const isEveryOptionSelected = options.length === selectedOptions.length
+
+  const handleClose = () => {
+    onClose?.()
+  }
+
+  return (
+    <BottomSheetBase
+      forwardedRef={forwardedRef}
+      onClose={handleClose}
+      onOpen={onOpen}
+      backgroundStyle={{ backgroundColor: 'transparent' }}
+      handleIndicatorStyle={{ width: 0 }}
+    >
+      <BottomSheetScrollView
+        forwardedRef={scrollViewRef}
+        testId={'MultiSelectBottomSheet'}
+        containerStyle={styles.bottomSheetScrollView}
+      >
+        <View style={[styles.option, styles.borderRadiusTop]}>
+          <Text style={styles.boldTextStyle}>{title}</Text>
+        </View>
+        <View style={styles.optionsContainer}>
+          <ScrollView style={{ maxHeight: OPTION_HEIGHT * MAX_OPTIONS_IN_VIEW }}>
+            <OptionLineItem
+              text={selectAllText}
+              isSelected={isEveryOptionSelected}
+              onPress={() => setSelectedOptions(options.map((option) => option.id))}
+            />
+            {options.map((option) => (
+              <OptionLineItem
+                text={option.text}
+                iconUrl={option.iconUrl}
+                isSelected={
+                  !!selectedOptions.find((selectedOption) => selectedOption === option.id) &&
+                  !isEveryOptionSelected
+                }
+                onPress={() => {
+                  if (isEveryOptionSelected) {
+                    setSelectedOptions([option.id])
+                  } else {
+                    if (selectedOptions.find((selectedOption) => selectedOption === option.id)) {
+                      setSelectedOptions(
+                        selectedOptions.filter((selectedOption) => selectedOption !== option.id)
+                      )
+                    } else {
+                      setSelectedOptions([...selectedOptions, option.id])
```

Comment:
could there be a race if 2 elements are clicked really fast? chatgpt suggests something like this
```ts
const toggleOption = (option) => {
  setSelectedOptions(prevSelectedOptions => {
    const isSelected = prevSelectedOptions.includes(option);

    if (isSelected) {
      return prevSelectedOptions.filter((item) => item !== option);
    } else {
      return [...prevSelectedOptions, option];
    }
  });
};

Reply:
Ah interesting! I've never used passing a function into the `setState` function as an updater.

Reply:
This actually forced a much nicer unit tests!